src/gui/model.rs

```diff
@@ -243,55 +317,27 @@ impl Greeter {
                 self.start_session(sender, info);
             }
             AuthStatus::InProgress => {
-                self.send_password(sender, password, info);
+                self.send_input(sender, input, info);
             }
             AuthStatus::NotStarted => {
                 self.create_session(sender, info);
             }
         };
     }
 
-    /// Send the entered password for logging in.
-    fn send_password(
-        &mut self,
-        sender: &ComponentSender<Self>,
-        password: String,
-        info: &UserSessInfo,
-    ) {
+    /// Send the entered input for logging in.
+    fn send_input(&mut self, sender: &ComponentSender<Self>, input: String, info: &UserSessInfo) {
+        // TODO we should be able to remove this thanks to the new looping handler
```

Comment:
Looking at the code, I agree with you. If you tested this, and it works, then feel free to go ahead and remove the input reset.

Reply:
Decided to keep it like this now that I changed the structure again a bit. And resetting the input does fit here, so I'll just leave it alone.